pkg/auth/dependency/sso/authInfo.go

```diff
@@ -0,0 +1,85 @@
+package sso
+
+type authHandler struct {
```

Comment:
The name `authHandler` looks so general, and I thought it is the base class of different provider, but it is not. I suggest you to refactor this struct.

By the convention of `skygear` as I know, to share common logic, it should be called something like `getAuthInfoRequest` if you put those param (code, scope and encodedState) in the struct body.

Also, it does not look so golang with those function variable, they can be easily replaced by interface.